thunder/core/profile.py

```diff
@@ -1,11 +1,28 @@
+from collections.abc import Callable
 import contextlib
+import functools
 import os
+import warnings
 
 import torch
 
 
 _ENABLED = os.getenv("THUNDER_ANNOTATE_TRACES") in ("1", "y", "Y")
 
+# However, nvtx is incredibly cheap so we no longer bother requiring the
+# environment variable.
+try:
+    import nvtx
+
+    _ENABLED = True
+except ImportError:
+    if _ENABLED:
+        msg = "Requested nvtx but the package is not available."
+        msg += "\nUse `pip install -m pip install nvtx`."
+        warnings.warn(msg)
+        _ENABLED = False
+        raise
```

Comment:
My goal here was that people do not need to learn about our environment variable to get useful information out of thunder, which I would argue is good UX when thunder is just one of hundreds of packages in use.

Interactively we discussed wanting users to see the error if they requested annotations but don't have the lib, hence the re-raise.